audio/src/lib.rs

```diff
@@ -29,8 +30,65 @@ pub use fetch::{
     READ_AHEAD_BEFORE_PLAYBACK_ROUNDTRIPS, READ_AHEAD_BEFORE_PLAYBACK_SECONDS,
     READ_AHEAD_DURING_PLAYBACK_ROUNDTRIPS, READ_AHEAD_DURING_PLAYBACK_SECONDS,
 };
+use std::fmt;
+
+pub enum AudioPacket {
+    Samples(Vec<i16>),
+    Bytes(Vec<u8>),
+}
+
+impl AudioPacket {
+    pub fn samples(&self) -> &[i16] {
+        match self {
+            AudioPacket::Samples(d) => d,
+            AudioPacket::Bytes(_d) => panic!("can't return bytes on samples"),
+        }
+    }
+
+    pub fn samples_mut(&mut self) -> &mut [i16] {
+        match self {
+            AudioPacket::Samples(d) => d,
+            AudioPacket::Bytes(_d) => panic!("can't return bytes on samples"),
+        }
+    }
+
+    pub fn bytes(&self) -> &[u8] {
+        match self {
+            AudioPacket::Bytes(d) => d,
+            AudioPacket::Samples(_d) => panic!("can't return samples on bytes"),
+        }
+    }
+
+    pub fn empty(&self) -> bool {
+        match self {
+            AudioPacket::Samples(d) => d.len() == 0,
+            AudioPacket::Bytes(d) => d.len() == 0,
+        }
+    }
```

Comment:
AFAIK, `is_empty` would be more usual in Rust. Vecs also have a function `is_empty` which you could use here. But that's bikeshedding.

Reply:
done

Reply:
Would you please also replace the `.len() == 0`? [Clippy would give a warning](https://rust-lang.github.io/rust-clippy/master/#len_zero) (and we have already enough of those warnings imo...)